config/management/genesis/src/validator_builder.rs

```diff
@@ -305,19 +307,53 @@ impl ValidatorBuilder {
             validator.storage_config.clone(),
         ));
 
-        let fullnode_identity = fullnode_network.identity_from_storage();
-        fullnode_network.identity = Identity::from_storage(
-            fullnode_identity.key_name,
-            fullnode_identity.peer_id_name,
-            SecureBackend::OnDiskStorage(validator.storage_config.clone()),
-        );
+        // By default we don't start a swarm with VFNs, so make sure the public fullnode endpoint
+        // really is publicly accessable
+
+        let fullnode_network_listen_address =
+            if let Some(template_fullnode_config) = config.full_node_networks.first() {
+                template_fullnode_config.listen_address.clone()
+            } else {
+                diem_config::utils::get_available_port_in_multiaddr(true)
+            };
+        let fullnode_network = NetworkConfig {
+            listen_address: fullnode_network_listen_address,
+            network_id: NetworkId::Public,
+            max_outbound_connections: 0,
+            identity: Identity::from_storage(
+                FULLNODE_NETWORK_KEY.to_owned(),
+                OWNER_ACCOUNT.to_owned(),
+                SecureBackend::OnDiskStorage(validator.storage_config.clone()),
+            ),
+
+            ..Default::default()
```

Comment:
Does `..` by itself not do default?

Reply:
no, it takes an arbitrary struct as input.